### ISSUE TITLE
Add support for mixed Hashing (RSAES-OAEP/SHA-256/MGF1-SHA-1‏)

### ISSUE THREAD

**Issue by trezsoft** [randombit/botan#109]
Although not recommended at some time OpenSSL supported mixed hashing, due to this several other libraries on Java and Javascript support the same mixed hashing. Currently in Botan only the recommended hashing is supported (i.e. both the same ). For backward compatibility it would be great to allow mixed hashing in Botan

For comparison, see: https://github.com/digitalbazaar/forge

``` javascript
// encrypt data with a public key using RSAES-OAEP/SHA-256/MGF1-SHA-1
// compatible with Java's RSA/ECB/OAEPWithSHA-256AndMGF1Padding
var encrypted = publicKey.encrypt(bytes, 'RSA-OAEP', {
  md: forge.md.sha256.create(),
  mgf1: {
    md: forge.md.sha1.create()
  }
});
```

**Comment by randombit** [comment]
While certainly feeling quite 'off', I don't see any obvious cryptographic problem with mixing hashes in OAEP, so I think it's reasonable to support for compatibility.

**Comment by randombit** [comment]
Sorry about the long wait, fixed now and this will be available in 2.5
